### PR TITLE
adding channel inspect for ship and kots, release update for ship

### PR DESCRIPTION
**Pull request by smithers1221**

### CODE REVIEW

File: client/channel.go [hunk 1]
```diff
@@ -41,12 +42,40 @@ func (c *Client) ListChannels(appID string) ([]types.Channel, error) {
...
+	appType, err := c.GetAppType(appID)
+	if err != nil {
+		return err
+	}
+
+	if appType == "platform" {
+		return c.PlatformClient.ArchiveChannel(appID, channelID)
+	} else if appType == "ship" {
+		// return c.ShipClient.ArchiveChannel(appID, channelID)
+		return nil
```
Comment (dexhorthy) [inline client/channel.go]: do you think if this finishes without an error, that the user will believe that the channel was archived?

File: pkg/kotsclient/channel.go [hunk 2]
```diff
@@ -130,6 +140,80 @@ func ArchiveChannel(appID string, channelID string) error {
...
+      repoFullName
+      branch
+      path
+    }
+    extraLintRules
+    created
+    updated
+    isDefault
+    isArchived
+    releases {
```
Comment (dexhorthy) [inline pkg/kotsclient/channel.go]: do we need to fetch every release on the channel?

File: pkg/shipclient/channel.go [hunk 3]
```diff
@@ -121,6 +131,82 @@ func ArchiveChannel(appID string, channelID string) error {
...
+        repoFullName
+        branch
+        path
+      }
+      extraLintRules
+      created
+      updated
+      isDefault
+      isArchived
+      releases {
```
Comment (dexhorthy) [inline pkg/shipclient/channel.go]: same as above, do for fetching channels, do we need to get every release that was published to the channel?

File: client/channel.go (hunk 1 above)
Comment (smithers1221) [reply client/channel.go]: good point - will add that to the 'sorry, come back later' list.

File: pkg/kotsclient/channel.go (hunk 2 above)
Comment (smithers1221) [reply pkg/kotsclient/channel.go]: @dex - see your point, but can we table that for another PR.  Currently pulling from what's on Vendorweb.

**Pull request merged**